gobblin-core/src/main/java/org/apache/gobblin/source/extractor/extract/restapi/RestApiConnector.java

```diff
@@ -192,8 +190,8 @@ private static boolean hasId(JsonObject json) {
         if (httpEntity != null) {
           EntityUtils.consume(httpEntity);
         }
-        if(httpResponse instanceof CloseableHttpResponse) {
-          this.closer.register((CloseableHttpResponse)httpResponse);
+        if(httpResponse instanceof Closeable) {
+          this.closer.register((Closeable)httpResponse);
```

Comment:
I just want to make sure those lines will be executed no matter what happens during the execution to avoid any resource leak. Will address the code format

Reply:
if you on the line directly following `httpResponse` assignment (or preferably, initialization) it's guaranteed to execute if and only if the `httpClient.execute` returned a result normally, w/o throwing.